### PR TITLE
feat: support pnpm package manager

### PR DESCRIPTION
**Pull request by sarya-godaddy**
## Summary
- add support for pnpm package manager

## Changelog
- added pnpm as a choice for package manager prompt in `gasket create`
- updated PackageManager with pnpm commands

## Test Plan
- updated unit tests
- tested by creating a gasket project with pnpm locally

> __NOTE__: I haven't tested this on windows

### CODE REVIEW

**Comment by mmason2-godaddy** [comment]
@sarya-godaddy Are you still actively working on this PR? The subject of `pnpm` came up today funny enough. We're going to be SPIKING-OUT this feature.

**Comment by sarya-godaddy** [comment]
@mmason2-godaddy Yes I can pick this back up. Just need to update the command line output and unit tests. I will get this ready for review by end of next week 03/01. [gdcorp-action-public-forks/pnpm-action-setup](https://github.com/gdcorp-action-public-forks/pnpm-action-setup) is now also available for using pnpm in GHA

File: packages/gasket-utils/lib/package-manager.js [hunk 1]
```diff
@@ -114,6 +130,19 @@ class PackageManager {
...
         env
       });
+    } else if (this.manager === 'pnpm') {
+      const argv = [cmd].concat(args);
+
+      //
+      // Support for the .npmrc configured via --npmconfig flag.
+      // pnpm does not have a "userconfig" CLI flag, it does however still
+      // support the npm_config_* environment variables for npm compatibility.
+      if (this.npmconfig) env.NPM_CONFIG_USERCONFIG = this.npmconfig;
```
Comment (agerard-godaddy) [inline packages/gasket-utils/lib/package-manager.js:140]: The `--npmconfig` flag is being removed in the next major version. See https://github.com/godaddy/gasket/pull/647

While this is not a breaking change, if there's no immediate need for it now, could we just save `pnpm` support for our next major version 7?

Comment (sarya-godaddy) [reply packages/gasket-utils/lib/package-manager.js:140]: I don't see a need for it. people can migrate from npm to pnpm pretty easily. I will update the PR to merge to v7 base

**Comment by sarya-godaddy** [comment]
there is a PR for releasing this to v7 https://github.com/godaddy/gasket/pull/797

**Pull request closed without merging**